### ISSUE TITLE
`core/navigation` block is breaking with on WP v6.5.5 & Block Visibility v3.5.0

### ISSUE THREAD

**Issue by up1512001** [ndiego/block-visibility#111]
## Description
I have a site running on `WP v6.5.5`, and before the release of `3.5.0` for block visibility, the navigation block was working as expected. However, after updating the plugin to `3.5.0`, the `core/navigation` block is breaking.

## Step-by-step reproduction instructions
- create black Wordpress local site.
- run this command to change WP version to 6.5.5 `wp core update --version=6.5.5 --force`
- activate TT4 or any default WP theme and activate `Block Visibility` plugin.
- create page and add `core/navigation` block on page then block will not render correctly.

## Expected behavior
- `core/navigation` block should display navigation links.

## Actual behavior
- its showing error of `This block has encountered an error and cannot be previewed.`

## Screenshots or screen recording (optional)

https://github.com/user-attachments/assets/1554bf05-df38-4bbc-bbc1-f28b11631345

## System information
```

### wp-core ###

version: 6.5.5
site_language: en_US
user_language: en_US
timezone: +00:00
permalink: /%postname%/
https_status: false
multisite: false
user_registration: 0
blog_public: 1
default_comment_status: open
environment_type: local
user_count: 1
dotorg_communication: true

### wp-paths-sizes ###

wordpress_path: /Users/utsav/Local Sites/github-stats-plugin/app/public
wordpress_size: 58.50 MB (61339056 bytes)
uploads_path: /Users/utsav/Local Sites/github-stats-plugin/app/public/wp-content/uploads
uploads_size: 0.00 B (0 bytes)
themes_path: /Users/utsav/Local Sites/github-stats-plugin/app/public/wp-content/themes
themes_size: 3.21 MB (3367998 bytes)
plugins_path: /Users/utsav/Local Sites/github-stats-plugin/app/public/wp-content/plugins
plugins_size: 6.11 MB (6410762 bytes)
database_size: 2.66 MB (2785280 bytes)
total_size: 70.48 MB (73903096 bytes)

### wp-active-theme ###

name: Twenty Twenty-Four (twentytwentyfour)
version: 1.2
author: the WordPress team
author_website: https://wordpress.org
parent_theme: none
theme_features: core-block-patterns, post-thumbnails, responsive-embeds, editor-styles, html5, automatic-feed-links, widgets-block-editor, block-templates
theme_path: /Users/utsav/Local Sites/github-stats-plugin/app/public/wp-content/themes/twentytwentyfour
auto_update: Disabled

### wp-themes-inactive (1) ###

Twenty Twenty-One: version: 2.3, author: the WordPress team, Auto-updates disabled

### wp-plugins-active (1) ###

Block Visibility: version: 3.5.0, author: Nick Diego, Auto-updates disabled

### wp-plugins-inactive (1) ###

Github Stats: version: 1.0, author: Utsav Patel, Renish Surani, Auto-updates disabled

### wp-media ###

image_editor: WP_Image_Editor_Imagick
imagick_module_version: 1808
imagemagick_version: ImageMagick 7.1.0-10 Q16 arm 2021-10-05 https://imagemagick.org
imagick_version: 3.7.0
file_uploads: 1
post_max_size: 1000M
upload_max_filesize: 300M
max_effective_size: 300 MB
max_file_uploads: 20
imagick_limits: 
	imagick::RESOURCETYPE_AREA: 32 GB
	imagick::RESOURCETYPE_DISK: 9.2233720368548E+18
	imagick::RESOURCETYPE_FILE: 7872
	imagick::RESOURCETYPE_MAP: 32 GB
	imagick::RESOURCETYPE_MEMORY: 16 GB
	imagick::RESOURCETYPE_THREAD: 1
	imagick::RESOURCETYPE_TIME: 9.2233720368548E+18
imagemagick_file_formats: 3FR, 3G2, 3GP, AAI, AI, APNG, ART, ARW, ASHLAR, AVI, AVIF, AVS, BGR, BGRA, BGRO, BMP, BMP2, BMP3, BRF, CAL, CALS, CANVAS, CAPTION, CIN, CIP, CLIP, CMYK, CMYKA, CR2, CR3, CRW, CUBE, CUR, CUT, DATA, DCM, DCR, DCRAW, DCX, DDS, DFONT, DNG, DOT, DPX, DXT1, DXT5, EPDF, EPI, EPS, EPS2, EPS3, EPSF, EPSI, EPT, EPT2, EPT3, ERF, EXR, FARBFELD, FAX, FF, FILE, FITS, FL32, FLV, FRACTAL, FTP, FTS, G3, G4, GIF, GIF87, GRADIENT, GRAY, GRAYA, GROUP4, GV, HALD, HDR, HEIC, HEIF, HISTOGRAM, HRZ, HTM, HTML, HTTP, HTTPS, ICB, ICO, ICON, IIQ, INFO, INLINE, IPL, ISOBRL, ISOBRL6, JNG, JNX, JPE, JPEG, JPG, JPS, JSON, K25, KDC, KERNEL, LABEL, M2V, M4V, MAC, MAP, MASK, MAT, MATTE, MEF, MIFF, MKV, MNG, MONO, MOV, MP4, MPC, MPEG, MPG, MRW, MSL, MSVG, MTV, MVG, NEF, NRW, NULL, ORA, ORF, OTB, OTF, PAL, PALM, PAM, PANGO, PATTERN, PBM, PCD, PCDS, PCL, PCT, PCX, PDB, PDF, PDFA, PEF, PES, PFA, PFB, PFM, PGM, PGX, PHM, PICON, PICT, PIX, PJPEG, PLASMA, PNG, PNG00, PNG24, PNG32, PNG48, PNG64, PNG8, PNM, POCKETMOD, PPM, PS, PS2, PS3, PSB, PSD, PTIF, PWP, RADIAL-GRADIENT, RAF, RAS, RAW, RGB, RGB565, RGBA, RGBO, RGF, RLA, RLE, RMF, RW2, SCR, SCT, SFW, SGI, SHTML, SIX, SIXEL, SPARSE-COLOR, SR2, SRF, STEGANO, SUN, SVG, SVGZ, TEXT, TGA, THUMBNAIL, TIFF, TIFF64, TILE, TIM, TM2, TTC, TTF, TXT, UBRL, UBRL6, UIL, UYVY, VDA, VICAR, VID, VIFF, VIPS, VST, WBMP, WEBM, WEBP, WMV, WPG, X3F, XBM, XC, XCF, XPM, XPS, XV, YAML, YCbCr, YCbCrA, YUV
gd_version: bundled (2.1.0 compatible)
gd_formats: GIF, JPEG, PNG, WebP, BMP, AVIF
ghostscript_version: 9.56.1

### wp-server ###

server_architecture: Darwin 23.6.0 arm64
httpd_software: nginx/1.16.0
php_version: 8.3.8 64bit
php_sapi: fpm-fcgi
max_input_variables: 4000
time_limit: 1200
memory_limit: 256M
max_input_time: 600
upload_max_filesize: 300M
php_post_max_size: 1000M
curl_version: 8.7.1 (SecureTransport) LibreSSL/3.3.6
suhosin: false
imagick_availability: true
pretty_permalinks: true
htaccess_extra_rules: false
current: 2024-08-30T18:32:06+00:00
utc-time: Friday, 30-Aug-24 18:32:06 UTC
server-time: 2024-08-30T18:32:04+00:00

### wp-database ###

extension: mysqli
server_version: 8.0.16
client_version: mysqlnd 8.3.8
max_allowed_packet: 16777216
max_connections: 151

### wp-constants ###

WP_HOME: undefined
WP_SITEURL: undefined
WP_CONTENT_DIR: /Users/utsav/Local Sites/github-stats-plugin/app/public/wp-content
WP_PLUGIN_DIR: /Users/utsav/Local Sites/github-stats-plugin/app/public/wp-content/plugins
WP_MEMORY_LIMIT: 40M
WP_MAX_MEMORY_LIMIT: 256M
WP_DEBUG: false
WP_DEBUG_DISPLAY: true
WP_DEBUG_LOG: false
SCRIPT_DEBUG: false
WP_CACHE: false
CONCATENATE_SCRIPTS: undefined
COMPRESS_SCRIPTS: undefined
COMPRESS_CSS: undefined
WP_ENVIRONMENT_TYPE: local
WP_DEVELOPMENT_MODE: undefined
DB_CHARSET: utf8
DB_COLLATE: undefined

### wp-filesystem ###

wordpress: writable
wp-content: writable
uploads: writable
plugins: writable
themes: writable


```

**Comment by ndiego** [comment]
Thanks for reporting @up1512001. I can replicate this issue in 6.5.5, but it does not occur in 6.6+. This looks to be a backward compatibility issue related to the [unified extensibility APIs](https://make.wordpress.org/core/2024/06/18/editor-unified-extensibility-apis-in-6-6/) in 6.6. I'll work on a fix for version 3.6.0, which will be coming out next week.

**Comment by up1512001** [comment]
@ndiego Thanks 👍

**Comment by ndiego** [comment]
Fixed in https://github.com/ndiego/block-visibility/commit/a0461c768e1bdab9325f8737e94c0ffb0ee3ce6c. This will be included in v3.6.0, which will be released in the middle of next week. Aiming for Wednesday 9/4.
